lms/djangoapps/course_goals/models.py

```diff
@@ -34,3 +38,17 @@ def __unicode__(self):
 
     class Meta:
         unique_together = ("user", "course_key")
+
+
+@receiver(models.signals.post_save, sender=CourseEnrollment, dispatch_uid="create_course_goal_on_enroll")
+def set_course_goal_verified(sender, instance, **kwargs):  # pylint: disable=unused-argument, invalid-name
+    """Set the course goal to certify when the user enrolls as a verified user. """
+    ## If the user enrolls in a verified state or upgrades, create a course goal
+    if instance.mode == 'verified':
+        add_course_goal(instance.user, instance.course_id, 'certify')
+
+
+@receiver(models.signals.post_delete, sender=CourseEnrollment, dispatch_uid="remove_course_goal_on_unenroll")
+def remove_course_goal_on_unenroll(sender, instance, **kwargs):  # pylint: disable=unused-argument, invalid-name
+    """Remove course goal on unenroll. """
+    remove_course_goal(instance.user, instance.course_id)
```

Comment:
If we are deleting data, we should have some audit trail.  I don't know if @marcotuts wants a delete event to match the added/updated event?